src/common/motor.rs

```diff
@@ -192,10 +204,18 @@ impl TryFrom<&Kind> for MotorPinsConfig {
                 }
             },
         };
-        let pwm = value
-            .get("pwm")?
-            .ok_or_else(|| AttributeError::KeyNotFound("pwm".to_string()))?
-            .try_into()?;
+        let pwm = match value.get("pwm") {
+            Ok(opt) => match opt {
```

Comment:
would a Option::map work here?

Reply:
I wanted to do this but the problem I was having was dealing with the fact that `try_into` returns a result and not the actual value

Reply:
I figured out how to do it, but do you find this clearer?
```rust
let pwm = value.get("pwm").map(
    |opt| opt.map(
        |val| i32::try_from(val)
    )
)?.transpose().or_else(|err| match err {
    AttributeError::KeyNotFound(_) => Ok(None),
    _ => {
        Err(err)
    }
})?;
```